### PR TITLE
Remove addComputedTypeAnnotations tree version in the Interning Checker

### PR DESCRIPTION
**Pull request by AndrewShf**
I think we don't have to refine the lhs to be `interned` at this point (the `addComputedTypeAnnotations` tree version), thus removing it. Flow refinement will take care of it later.

But the element version is needed for bytecode analysis, as for bytecode it only gets the element type.

### CODE REVIEW

**Comment by wmdietl** [comment]
Thanks for fixing this! The Daikon failure seems unrelated.
Could you add a test case with a field, which is a compile time constant, and make sure the Interning Checker doesn't raise a redundant annotation warning?

**Comment by AndrewShf** [comment]
Oh, forgot to update a testcase...

File: checker/tests/interning-warnredundantannos/StaticFinalStringDefault.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import org.checkerframework.checker.interning.qual.Interned;
+
+public class StaticFinalStringDefault {
+    // the default type of str should not be @Interned, even though it's later refined to it.
```
Comment (wmdietl) [inline checker/tests/interning-warnredundantannos/StaticFinalStringDefault.java:4]:
```suggestion
    // The default type of str is not @Interned, even though it is later refined to it.
```

File: checker/src/test/java/org/checkerframework/checker/test/junit/InterningWarnRedundantAnnosTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        super(
+                testFiles,
+                org.checkerframework.checker.interning.InterningChecker.class,
+                "interning-warnredundantannos",
+                "-AwarnRedundantAnnotations");
+    }
+
+    @Parameterized.Parameters
+    public static String[] getTestDirs() {
+        return new String[] {"interning-warnredundantannos"};
```
Comment (wmdietl) [inline checker/src/test/java/org/checkerframework/checker/test/junit/InterningWarnRedundantAnnosTest.java:27]: Throughout, can you expand `annos` to `annotations`, to be uniform with the existing `nullness-warnredundantannotations` test names?

**Review by wmdietl — COMMENTED** [review]
Thanks, this fixes the daikon interning failures.
I'll work on a separate PR to fix the remaining daikon failure.

File: checker/src/test/java/org/checkerframework/checker/test/junit/InterningWarnRedundantAnnosTest.java (hunk 2 above)
Comment (AndrewShf) [reply checker/src/test/java/org/checkerframework/checker/test/junit/InterningWarnRedundantAnnosTest.java]: Fixed!

**Review by wmdietl — APPROVED** [review]
Thanks!

**Pull request merged**